### PR TITLE
[GRAPHBOLT][WARNINGS]Fixing warnings appearing in some `graphbolt` tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

The following warnings appearing in `graphbolt` tests have been fixed as suggested:
```
tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py: 17 warnings
tests/python/pytorch/graphbolt/test_item_sampler.py: 8 warnings
tests/python/pytorch/graphbolt/test_itemset.py: 1 warning
  /usr/local/lib/python3.10/dist-packages/dgl/graphbolt/itemset.py:182: UserWarning: To copy construct from a tensor, 
it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), 
rather than torch.tensor(sourceTensor).
    return torch.tensor(index, dtype=dtype)
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
```
FAILED tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py::test_OnDiskDataset_all_nodes_set_homo - AttributeError: 'int' object has no attribute 'type'
FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_scalar_dtype[dtype0] - AttributeError: 'int' object has no attribute 'type'
FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_scalar_dtype[dtype1] - AttributeError: 'int' object has no attribute 'type'
FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_length - AttributeError: 'int' object has no attribute 'type'
FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_seed_nodes - AttributeError: 'int' object has no attribute 'type'
=== 5 failed, 6077 passed, 1088 skipped, 5147 warnings in 703.85s (0:11:43) ===
```

**Comment by frozenbugs** [comment]
@dgl-bot

File: python/dgl/graphbolt/itemset.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def __getitem__(self, index: Union[int, slice, Iterable[int]]):
                     )
                 return torch.tensor(index, dtype=dtype)
             elif isinstance(index, Iterable):
-                return torch.tensor(index, dtype=dtype)
+                return index.type(dtype).clone().detach()
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/itemset.py:181]: 1. if index is a PyTorch tensor, could we return `index.to(dtype)`?
2. Ideally we need change `elif isinstance(index, Iterable):` to `elif isinstance(index, torch.Tensor):`

Comment (drivanov) [reply python/dgl/graphbolt/itemset.py]: Done

**Comment by Rhett-Ying** [comment]
@dgl-bot

File: python/dgl/graphbolt/itemset.py [hunk 2]
```diff
@@ -177,8 +177,8 @@ def __getitem__(self, index: Union[int, slice, Iterable[int]]):
                         f"{type(self).__name__} index out of range."
                     )
                 return torch.tensor(index, dtype=dtype)
-            elif isinstance(index, Iterable):
-                return torch.tensor(index, dtype=dtype)
+            elif isinstance(index, torch.Tensor):
+                return index.to(dtype)
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/itemset.py:181]: line 185 should be updated accordingly as well.

Comment (drivanov) [reply python/dgl/graphbolt/itemset.py:181]: Done.

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@drivanov 
```
FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_seed_nodes - AssertionError: Regex pattern did not match.

 Regex: "ItemSet indices must be int, slice, or iterable of int, not <class 'float'>."

 Input: "ItemSet indices must be int, slice, or torch.Tensor, not <class 'float'>."

```

**Comment by drivanov** [comment]
> @drivanov
> 
> ```
> FAILED tests/python/pytorch/graphbolt/test_itemset.py::test_ItemSet_seed_nodes - AssertionError: Regex pattern did not match.
> 
>  Regex: "ItemSet indices must be int, slice, or iterable of int, not <class 'float'>."
> 
>  Input: "ItemSet indices must be int, slice, or torch.Tensor, not <class 'float'>."
> ```

Per the reviewer's request, ([Rhett-Ying](https://github.com/Rhett-Ying) [last week](https://github.com/dmlc/dgl/pull/7582#discussion_r1724142865))
```
line 185 should be updated accordingly as well.
```
I have [updated](https://github.com/dmlc/dgl/pull/7582/commits/0f126be18d8fa01b278e13474a548acf50d82dc7) the error message. Therefore, **Regex** and **Input** no longer match. **Regex** needs to be updated accordingly.

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**